buildcontext/git.go

```diff
@@ -66,6 +66,39 @@ type resolvedGitProject struct {
 	state pllb.State
 }
 
+func (gr *gitResolver) expandWildcardPath(ctx context.Context, gwClient gwclient.Client, platr *platutil.Resolver, target domain.Target, pattern string) ([]string, error) {
+	if !target.IsRemote() {
+		return nil, errors.Errorf("unexpected local reference %s", target.String())
+	}
+
+	rgp, _, subDir, err := gr.resolveGitProject(ctx, gwClient, platr, target)
+	if err != nil {
+		return nil, err
+	}
+
+	gitRef, err := llbutil.StateToRef(ctx, gwClient, rgp.state, false, platr.SubResolver(platutil.NativePlatform), nil)
```

Comment:
I also tried to add `llb.Run(llb.Shlex("ls <path"))` here but I couldn't get it to work. The code here works well, but is limited to simple wildcard paths that end with `*` like `my/path/*`.

Reply:
Here's the error. I found this surprising because the base image is `alpine/git:v2.43.0` and appears to include `ls`.

```
            internal *failed* | Repeating the failure error...
            internal *failed* | runc run failed: unable to start container process: exec: "/bin/sh": stat /bin/sh: no such file or directory
            internal *failed* | ERROR
            internal *failed* |       The internal command
            internal *failed* |           /bin/sh -c ls ./subdir/*
            internal *failed* |       did not complete successfully. Exit code 1
```

Reply:
Hmm.. not sure why that happens.

It may be worth trying to get that to work, since supporting patterns like `BUILD ./cmd-*+binary` would be useful.

Reply:
it might be a bit dirty (since it's no longer a git command), but maybe we could just add another line to https://github.com/earthly/earthly/blob/f51a0c1cbbbb44520d5343b373e66d8004f5ee75/buildcontext/git.go#L237

which would run a `find -name Earthfile > /dest/Earthfile-paths`, then a corresponding `gitMetaRef.ReadFile` which would return the list of Earthfiles in a new field under the `resolvedGitProject` struct.

Reply:
That's actually pretty reasonable IMO

Reply:
Good idea. This would help remove the new code path in `gitResolver` & remove the need to run a custom `ls` or `find` command.